### PR TITLE
Add support to install `yq` from GitHub

### PR DESCRIPTION
**Pull request by Divyansh42**
Signed-off-by: divyansh42 <diagrawa@redhat.com>

### Related Issue(s)
Closes: #30 

### Checklist

- [x] This PR includes a documentation change
- [ ] This PR does not need a documentation change
---
- [x] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is a patch change
- [x] This change is a minor change
- [ ] This change is a major (breaking) change

### CODE REVIEW

File: src/installer/hash.ts [hunk 1]
```diff
@@ -118,7 +118,14 @@ async function getOnlineHash(clientFile: ClientFile): Promise<HashFileContents |
     let hash;
 
     if (clientFile.clientName === Inputs.YQ) {
-        hash = parseHashFileForYq(hashFileContents, clientFile.archiveFilename);
+        const checksumHashesOrderfileUrl = getGitHubReleaseAssetPath(
```
Comment (tetchel) [inline src/installer/hash.ts:121]: can you capture the entire yq hash fetching into a function

File: src/installer/hash.ts [hunk 2]
```diff
@@ -152,20 +159,24 @@ function parseHashFile(hashFileContents: string, fileToHash: string): string {
  * Since yq has checksum file in different format from other clients.
  * @returns The hash for fileToHash, as extracted from the hashFileContents.
  */
-function parseHashFileForYq(hashFileContents: string, fileToHash: string): string {
+function parseHashFileForYq(hashFileContents: string, checksumHashesOrdercontents: string, fileToHash: string): string {
+
+    const checksumHashesOrder = checksumHashesOrdercontents.split("\n");
```
Comment (tetchel) [inline src/installer/hash.ts:164]: can you link to an existing release's hash file and orders file here so another developer can easily find the file you are trying to parse

File: src/util/file-finder.ts [hunk 3]
```diff
@@ -55,6 +56,18 @@ export async function findMatchingClient(source: string, client: InstallableClie
         ghCore.debug(`${client} ${clientVersion} files: ${clientFiles.join(", ")}`);
     }
 
+    // In case of client being 'yq', executable and zip files both are present
+    // this will give warning to the users as multiple files will be found after
+    // filteration. So removing all the zipped files.
+    if (client === Inputs.YQ) {
+        clientFiles.forEach((file) => {
```
Comment (tetchel) [inline src/util/file-finder.ts:63]: this is a poor way of filtering.

To keep with the way the rest of the filters work, I'd add a filter to the filter pipelines below that is the opposite of `filterByZipped`, ie `filterByNotZipped`.

If I wanted to do it outside of the filter pipeline, I would do:

`clientFiles = clientFiles.filter((file) => !canExtract(file))`

**Review by tetchel — CHANGES_REQUESTED** [review]

File: src/installer/hash.ts (hunk 2 above)
Comment (Divyansh42) [reply src/installer/hash.ts:164]: Okay :+1:

File: src/util/file-finder.ts (hunk 3 above)
Comment (Divyansh42) [reply src/util/file-finder.ts:63]: Yeah, this approach is far better. I will change that

**Review by tetchel — APPROVED** [review]

**Pull request merged**